Src/Scripts/Include-GenerateUnitClassSourceCode.ps1

```diff
@@ -278,11 +280,30 @@ namespace UnitsNet
         public static $className Parse(string str, IFormatProvider formatProvider = null)
         {
             if (str == null) throw new ArgumentNullException("str");
-            string[] words = str.Split(new[] {" "}, StringSplitOptions.RemoveEmptyEntries);
-            if (words.Length < 2)
+
+            var numFormat = formatProvider != null ?
+                (NumberFormatInfo) formatProvider.GetFormat(typeof (NumberFormatInfo)) :
+                (NumberFormatInfo) CultureInfo.CurrentCulture.NumberFormat.Clone();
```

Comment:
I don't know if you considered CurrentCulture vs CurrentUICulture here? I'm not sure which one is right.

http://stackoverflow.com/a/329041/134761
> CurrentCulture is the .NET representation of the default user locale of the system. This controls default number and date formatting and the like.
> 
> CurrentUICulture refers to the default user interface language, a setting introduced in Windows 2000. This is primarily regarding the UI localization/translation part of your app.

I've used CurrentUICulture as default in the unit classes' ToString() and in UnitSystem for outputting and parsing unit abbreviations, since I figured that was the most closely related to localization and text presentation in an app. 

I don't know if CurrentCulture or CurrentUICulture is best suited for parsing quantities in this context? Normally the two are similar, but there is a reason they are separate. The scenario I can best relate to is if an app allows me to choose language, then I usually choose English. I haven't done much localization myself, but I assume it then sets the Thread.CurrentUICulture to "en-US" so those text resources are loaded etc. But being Norwegian, I prefer reading Norwegian dates and numbers so I have configured my Windows to show that. So I'm thinking Thread.CurrentCulture might remain "nb-NO" in a properly localized app, so it shows and parses numbers and dates in my preferred format. If that is the case, then CurrentCulture should be used in this line.

Thoughts?
Sorry for the big post, but I'm fuzzy on what's best practice here.